runtime/compiler/optimizer/J9TransformUtil.cpp

```diff
@@ -1797,17 +1939,33 @@ J9::TransformUtil::transformIndirectLoadChain(TR::Compilation *comp, TR::Node *n
  *
  */
 bool
-J9::TransformUtil::transformIndirectLoadChainImpl(TR::Compilation *comp, TR::Node *node, TR::Node *baseExpression, void *baseAddress, int32_t baseStableArrayRank, TR::Node **removedNode)
+J9::TransformUtil::transformIndirectLoadChainImpl(TR::Compilation *comp,
+                                                  TR::Node *node, TR::Node *baseExpression,
+                                                  TR::KnownObjectTable::Index baseKnownObject,
+                                                  void *baseAddress,
+                                                  int32_t baseStableArrayRank,
+                                                  TR::Node **removedNode)
    {
-   bool isBaseStableArray = baseStableArrayRank > 0;
+#if defined(J9VM_OPT_JITSERVER)
+   bool isServer = comp->isOutOfProcessCompilation();
+
+   // baseKnownObject is used only for jitserver, otherwise we use baseAddress
+   if (isServer)
+      TR_ASSERT(baseKnownObject != TR::KnownObjectTable::UNKNOWN,
+                "invalid baseKnownObject in jitserver");
+   else
+#endif /* defined(J9VM_OPT_JITSERVER) */
+      TR_ASSERT(baseAddress, "invalid base address");
```

Comment:
On this "else" part we can add the original check:
`TR_ASSERT(TR::Compiler->vm.hasAccess(comp), "transformIndirectLoadChain requires VM access");`